Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -285,8 +285,8 @@ def warn_about_non_apple_gcc(name)
       gcc_formula = gcc_version_formula(name)
     rescue FormulaUnavailableError => e
       raise <<-EOS.undent
-      Homebrew GCC requested, but formula #{e.name} not found!
-      You may need to: brew tap homebrew/versions
+      Compiler #{e.name.sub("@", "-")} requested, but formula #{e.name} not found!
```

Comment:
Would this ever be not GCC? If not I'd leave the old message but just add the name/version.